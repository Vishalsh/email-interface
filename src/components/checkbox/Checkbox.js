import React from 'react';
import PropTypes from 'prop-types';

import classes from './Checkbox.module.scss';

const Checkbox = (props) => {
  const {
    labelClass,
    labelFor,
    onChange
  } = props;

  return (
    <>
    <input id={`checkbox_${labelFor}`}
           className={classes.input}
           type="checkbox"
           onChange={onChange}/>
    <label className={`${classes.label} ${labelClass}`}
           htmlFor={`checkbox_${labelFor}`}/>
    </>
  );
};

Checkbox.defaultProps = {
  labelClass: ''
};

Checkbox.propTypes = {
  labelClass: PropTypes.string,
  labelFor: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Checkbox;
