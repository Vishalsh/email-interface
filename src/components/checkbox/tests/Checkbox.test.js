import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Checkbox from '../Checkbox';

describe('Checkbox', () => {
  const props = {
    labelClass: 'labelClass',
    labelFor: '1',
    onChange: jest.fn(),
  };
  const component = shallow(<Checkbox {...props} />);

  it('should render the component', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
});