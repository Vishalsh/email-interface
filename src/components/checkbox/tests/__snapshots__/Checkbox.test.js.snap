// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render the component 1`] = `
<Fragment>
  <input
    className="input"
    id="checkbox_1"
    onChange={[MockFunction]}
    type="checkbox"
  />
  <label
    className="label labelClass"
    htmlFor="checkbox_1"
  />
</Fragment>
`;
